### PR TITLE
refactor(port): move from cloud APIs to evpn-gw APIs

### PR DESCRIPTION
**Pull request by glimchb**
Signed-off-by: Boris Glimcher <Boris.Glimcher@emc.com>

### CODE REVIEW

**Comment by glimchb** [comment]
@mardim91 I used this example
```
########### Setup Bridge Ports ##############
ip link set eth1 master br-tenant up
bridge vlan add dev eth1 vid 10 pvid untagged
ip link set eth2 master br-tenant up
bridge vlan add dev eth2 vid 20 pvid untagged
ip link set eth3 master br-tenant up
bridge vlan add dev eth3 vid 40
bridge vlan add dev eth3 vid 50
```

File: pkg/evpn/port.go [hunk 1]
```diff
@@ -27,117 +25,114 @@ import (
...
-		return nil, err
-	}
-	// set MAC
-	if len(ifaceobj.MacAddress) > 0 {
-		// TODO: dummy or vlandev ?
-		if err := netlink.LinkSetHardwareAddr(dummy, ifaceobj.MacAddress); err != nil {
-			fmt.Printf("Failed to set MAC on link: %v", err)
+		// Example: ip link set eth2 master br-tenant
+		if err := netlink.LinkSetMaster(iface, bridgeDev); err != nil {
+			fmt.Printf("Failed to add iface to bridge: %v", err)
```
Comment (mardim91) [inline pkg/evpn/port.go:117]: I think this is not right. Not sure what are you trying to do here

**Review by mardim91 — COMMENTED** [review]
I guess this is just the structure code for moving from cloud API to EVPN GW API. I have made some comments and also definetly some things are not working right. Please check the comments and if you want to merge it as we did in LogicalBridge PR then do so

File: pkg/evpn/port.go [hunk 2]
```diff
@@ -27,117 +25,114 @@ import (
...
-	if ifaceobj.VpcNameRef != "" {
-		// Validate that a Subnet/Bridge resource name conforms to the restrictions outlined in AIP-122.
-		if err := resourcename.Validate(ifaceobj.VpcNameRef); err != nil {
-			log.Printf("error: %v", err)
+	switch in.BridgePort.Spec.Ptype {
+	case pb.BridgePortType_ACCESS:
+		// TODO: fixme
+		fmt.Println("ACCESS")
+	case pb.BridgePortType_TRUNK:
+		// Example: bridge vlan add dev eth2 vid 20 pvid untagged
```
[the remark this answers is not in the document]
Comment (glimchb) [reply pkg/evpn/port.go:87]: @mardim91 access vs trunk - we have type - ok

but how to determine if no vlans (Transparent trunks) ?

File: pkg/evpn/port.go (hunk 1 above)
Comment (glimchb) [reply pkg/evpn/port.go]: see now better ?

File: docker-compose.yml [hunk 3]
```diff
@@ -293,13 +299,17 @@ services:
       /entrypoint.sh call --json_input --json_output localhost:50151 CreateLogicalBridge    "{\"logical_bridge\" : {\"spec\" : {\"vni\" : 11, \"vlan_id\": 11 }}, \"logical_bridge_id\" : \"vxlan11\" }" && \
       /entrypoint.sh call --json_input --json_output localhost:50151 CreateLogicalBridge    "{\"logical_bridge\" : {\"spec\" : {\"vni\" : 22, \"vlan_id\": 22 }}, \"logical_bridge_id\" : \"vxlan22\" }" && \
       /entrypoint.sh call --json_input --json_output localhost:50151 CreateLogicalBridge    "{\"logical_bridge\" : {\"spec\" : {\"vni\" : 33, \"vlan_id\": 33 }}, \"logical_bridge_id\" : \"vxlan33\" }" && \
+      /entrypoint.sh call --json_input --json_output localhost:50151 CreateBridgePort    "{\"bridge_port\" : {\"spec\" : {\"mac_address\" : \"qrvMAAAB\", \"ptype\": \"ACCESS\", \"logical_bridges\": [] }}, \"bridge_port_id\" : \"eth2\" }" && \
+      /entrypoint.sh call --json_input --json_output localhost:50151 CreateBridgePort    "{\"bridge_port\" : {\"spec\" : {\"mac_address\" : \"qrvMAAAB\", \"ptype\": \"TRUNK\" , \"logical_bridges\": [] }}, \"bridge_port_id\" : \"eth3\" }" && \
```
[the remark this answers is not in the document]
Comment (glimchb) [reply docker-compose.yml:303]: will fill this after #76 is ready

**Comment by glimchb** [comment]
will merge as infrastructure move to new APIs, all feedback and review on the content of the implementation is still very valid and welcome

File: pkg/evpn/port.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (mardim91) [reply pkg/evpn/port.go]: You check if the list of LogicalBridges [here](https://github.com/opiproject/opi-api/blob/main/network/evpn-gw/v1alpha1/l2_xpu_infra_mgr.proto#L237) is empty if it is then we have a transparent trunk bridgePort if not then it is a selective trunk and you need to configre Vlans.

Also for Access bridgePort type the LogicalBridge list must have only one item. If not then we need to throw error

**Pull request merged**